packages/core/src/render3/node_manipulation.ts

```diff
@@ -805,75 +743,85 @@ function appendProjectedNode(
  *               complication is that the nodes we are projecting can themselves have Containers
  *               or other Projections.
  *
- * As you can see this is a very recursive problem. While the recursive implementation is not the
- * most efficient one, trying to unroll the nodes non-recursively results in very complex code that
- * is very hard (to maintain). We are sacrificing a bit of performance for readability using a
- * recursive implementation.
+ * As you can see this is a very recursive problem. Yes recursion is not most efficient but the
+ * code is complicated enough that trying to implemented with recursion becomes unmaintainable.
  *
  * @param renderer Renderer to use
  * @param action action to perform (insert, detach, destroy)
  * @param lView The LView which needs to be inserted, detached, destroyed.
  * @param renderParent parent DOM element for insertion/removal.
  * @param beforeNode Before which node the insertions should happen.
  */
-function executeActionOnView(
+function applyView(
     renderer: Renderer3, action: WalkTNodeTreeAction, lView: LView, renderParent: RElement | null,
-    beforeNode: RNode | null | undefined) {
+    beforeNode: RNode | null) {
   const tView = lView[TVIEW];
   ngDevMode && assertNodeType(tView.node !, TNodeType.View);
-  let viewRootTNode: TNode|null = tView.node !.child;
-  while (viewRootTNode !== null) {
-    executeActionOnNode(renderer, action, lView, viewRootTNode, renderParent, beforeNode);
-    viewRootTNode = viewRootTNode.next;
-  }
+  const viewRootTNode: TNode|null = tView.node !.child;
+  applyNodes(renderer, action, viewRootTNode, lView, renderParent, beforeNode, false);
 }
 
 /**
- * `executeActionOnProjection` performs an operation on the projection specified by `action`
- * (insert, detach, destroy).
+ * `applyProjection` performs operation on the projection.
  *
  * Inserting a projection requires us to locate the projected nodes from the parent component. The
- * complication is that those nodes themselves could be re-projected from their parent component.
+ * complication is that those nodes themselves could be re-projected from its parent component.
```

Comment:
Revert? Think `their` is correct

```suggestion
 * complication is that those nodes themselves could be re-projected from their parent component.
```